Dev

NikiPshg opened the pull request:
- recovery from meta
 - saving meta in parquet
 - code refactoring
 - minor bug fixes

NikiPshg committed:
init

NikiPshg committed:
feat(example): add BalalaikaDataset demo class

- Add requirements.txt
- Save metadata to parquet format
- Implement podcast download using existing metadata
- Add example dataset class in example/dataset.py

NikiPshg committed:
fix: rename classificaton → classification and fix bugs in phonemizer/yofication

- Rename directory  →  including all internal files
- Fix typos in module paths and imports accordingly
- Add empty models folder with .gitignore to preserve structure
- Fix bug in src/phonemizer/phonemizer.py
- Fix bug in src/yofication/yofication.py

NikiPshg committed:
Merge main into dev

KORALLLL commented on configs/config.yaml:
На данный момент мы работаем с несколькими конфигурациями датасета. Возможные конфигурации - 100 часов, 500 часов, 1000 часов, 2000 часов. Может быть рассмотрим большие. Но пока что нет.

Назови этот конфиг так, чтобы было понятно какая это конфигурация датасета.

плюсом, ты говорил, что альбомы могут обновиться, а у нас текста в мете может не быть. Получится, что юзер зря скачивал подкаст. И ты предлагал указывать ссылки на конкретные записи. Как ты на это сейчас смотришь? Жду ответа в тг

Плюсом, мб если у нас всё фиксировано в датасете и разным будет только подкасты, которые нам надо скачать, мб оставить этот файл без изменений, а ссылки в отдельный файл добавить? Я хочу минимизировать повторяющиеся фрагменты

KORALLLL commented on example/dataset.py:
Можно редми по этому файлу? Меня интересуют следующие вопросы
1. когда он используется?
2. правильно ли я понимаю, что подкасты в него приходят уже нарезанными?
3. мета читатся из одного .parquet файла? Какой размер .parquet файла для 100 часов выходит? 

По поводу 3. Я боюсь, что при большом количестве меты мы не сможем открыть из просту файл с метой где угодно. И тогда у нас будет несколько файлов. Если я прав, то надо будет произвести замену с pandas на polars. И поларсом не читать, а стримингово сканить

NikiPshg replied:
Это метакласс датасета. 
Да подкасты уже нарезанные нужно только мета к ним
Пока что да для 100 часов 40Мб

KORALLLL commented on example/example.py:
для файла `example/example.py` я запросил редми. Можно ещё, пожалуйста, в редми добавить описание того, что на выходе из твоего датасета мы имеем? Мб это редми можно сделать не в этой папке, а добавить в общее редми репозитория

KORALLLL commented on models/.gitignore:
нужен ли этот файл тут? мб просто в не вложенный .gitignore его добавить, если надо игнорировать эту папку

NikiPshg replied:
папка models нужна для запуска пайплайна можно ее и убрать тогда юзер должен mkdir сделать

KORALLLL replied:
На будущее, все, тчо надо игнорировать, надо занести в один гит игнор. не надо в разных местах разные гит игноры

KORALLLL replied:
убери лишние гитигноры

KORALLLL commented on src/accents/README.md:
```diff
@@ -0,0 +1,45 @@
+- `--podcasts_path`: Root directory containing the text files for accent restoration (default: "../../../podcasts").
+- `--num_workers`: Number of worker processes per GPU for parallel processing (default: 4).
+- `--model_name`: Model version to use with RUAccent (default: "turbo3.1").
+- `--device`: Device to run the model on (default: "cuda").
```
вопрос по строке с выбором девайса. Реализована ли в коде возможность работы одновременно на нескольких гпу? мб запустить два параллельных процесса на двух гпу? Если нет, создай issue на этот счёт с лейблом `feature`

KORALLLL replied:
актуально

NikiPshg replied:
да код использует все доступные ему ресурсы и распределяет нагрузку равномерно (1, 1) , (2, 2 )....

NikiPshg committed:
feat: add multi-size dataset download system

• Added 100h/500h/2000h download scripts
• Implemented download_prepared for reproducibility
• Updated config system and documentation

NikiPshg committed:
update readme

KORALLLL commented on configs/.gitignore:
убрать

KORALLLL commented on configs/config.yaml:
```diff
@@ -0,0 +1,96 @@
+  episodes_limit: 8
+  num_workers: 2
+  parquet_path: /home/nikita/balalaika/balalaika.parquet
+  podcasts_urls:  
```
комментарии по этому файлу я добавил, но, почему, он "добавлен заново", хотя выглядит, что он просто изменён

KORALLLL commented on example/example.py:
```diff
@@ -0,0 +1,22 @@
+    dataset = BalalaikaDataset(
+        podcasts_path='../balalaika2',
+        parquet_path='../balalaika/balalaika.parquet'
+    )
```
тоже выглядит так, будто он новый, хотя он изменён. Старайся избегать копипасти, потому что это очень неудобно отображается в гите

KORALLLL commented on src/accents/accents.py:
```diff
@@ -0,0 +1,158 @@
+
+
+def process_file(path: Path) -> None:
+    if str(path).endswith('_accent.txt') or str(path).endswith('_giga.txt') or str(path).endswith('_giga.txt'):
```
```python
str(path).endswith('_giga.txt') or str(path).endswith('_giga.txt')
```

это же вроде всегда вернёт 1 или я что-то не поинмаю

KORALLLL replied:
а зачем ты всё условие убрал? Там надо было ток повтоярющееся условие убрать. Или в этом есть какяа-то логика?

NikiPshg replied:
строки 50 - 58 функция возвращает ссылки только с префиксом 
<img width="1528" height="374" alt="изображение" src="https://github.com/user-attachments/assets/293d98e9-a368-44c4-ae15-e979e82beb37" />
нет кейса где эти ифы отработают ( можно считать рудементом прошлой реализации)

KORALLLL commented on src/accents/accents.py:
```diff
@@ -0,0 +1,158 @@
+    config = load_config(args.config_path, 'accent')
+    num_workers = args.num_workers if args.num_workers else config.get('num_workers', 4)
+    model_name = args.model_name if args.model_name else config.get('model_name', 'turbo3.1')
+    device = args.device if args.device else config.get('device', 'cuda')
```
переменная device нигде не используется. Её либо убрать, либо использовать. Конечно, хотелось бы использовать, НО, я думаю сейчас можно забить и добавить задачу в беклог + создать issue на гитхабе, а потом уже сделаем поддержку

NikiPshg replied:
тут дело в том что ruaccent сам по себе не принимает device поэтому девайс там больше на будущее

KORALLLL commented on src/classification/.gitignore:
```diff
@@ -0,0 +1,4 @@
+voxblink2_samresnet100_ft/*
+chromadb
+cl_*.py
+*.ipynb
```
Пж, не создавай внутри папок гитигноры, используй один не вложенный, который у тебя есть

NikiPshg replied:
хорошо

KORALLLL commented on src/classification/classification_yaml.sh:
```diff
@@ -0,0 +1,22 @@
+
+activate_venv ".support_venv"
+
+python3 -m src.classification.classification --config "$CONFIG_PATH"
```
в питоновском файле он называется `config_path`

NikiPshg replied:
верно

KORALLLL commented on src/download/download_prepared.sh:
```diff
@@ -0,0 +1,21 @@
+
+SCRIPT_DIR=$(dirname "$(realpath "$0")")
+
+python3 -m src.download.download_prepared --pickle_path "$PICKLE_PATH"  --podcasts_path "$PODCASTS_PATH"
```
`--num_workers`

KORALLLL commented on src/phonemizer/phonemizer.py:
```diff
@@ -0,0 +1,129 @@
+import argparse
+import os
```
не нужный импорт

NikiPshg replied:
исправим

KORALLLL commented on src/phonemizer/phonemizer.py:
```diff
@@ -0,0 +1,129 @@
+import os
+from pathlib import Path
+from typing import Any, List
+import yaml
```
не нужный импорт

KORALLLL commented:
general: надо написать наверное в обычном редми, что если device это куда. то надо без цифр указывать, потому что у тебя автоматический асайнмент

KORALLLL commented on src/preprocess/preprocess.py:
```diff
@@ -0,0 +1,327 @@
+
+import torch
+import torchaudio
+import yaml
```
не используемый импорт

KORALLLL commented on src/separation/README.md:
```diff
@@ -0,0 +1,42 @@
+
+## Explanation of Parameters
+
+- `--config_path`: Path to the YAML configuration file.
+- `--podcasts_path`: Root directory containing podcast audio files for processing.
+- `--one_speaker`: Boolean flag to indicate if only one speaker is expected per audio file (default: True).
+- `--num_workers`: Number of parallel processes for audio processing (default: 4).
```
Не все параметры объяснены

NikiPshg replied:
объяснены ключевые параметры остальное понятно и так

KORALLLL replied:
Оно понятно, но хорошим тоном является описание. Лучше его добавить

KORALLLL commented on src/separation/separation.py:
```diff
@@ -0,0 +1,299 @@
+import argparse
+import os
+import sys
```
не используемый импорт

KORALLLL commented on src/separation/separation.py:
```diff
@@ -0,0 +1,299 @@
+import multiprocessing
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
```
`Any` и `Tuple` не используются

KORALLLL commented on src/separation/separation.py:
```diff
@@ -0,0 +1,299 @@
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+import yaml
+import shutil
```
не используется

KORALLLL commented on src/separation/separation.py:
```diff
@@ -0,0 +1,299 @@
+import pandas as pd
+import torch
+import torchaudio
+from dotenv import load_dotenv
```
не используется

KORALLLL commented on src/separation/separation.py:
```diff
@@ -0,0 +1,299 @@
+from loguru import logger
+from pyannote.audio import Pipeline
+from tqdm import tqdm
+from transformers import pipeline 
```
не используется

KORALLLL commented on src/transcription/README.md:
```diff
@@ -0,0 +1,35 @@
+
+## Explanation of Parameters
+
+- `--config_path`: Path to the YAML configuration file.  
```
т.е. задавать девайс можно через конфиг

NikiPshg replied:
да

KORALLLL commented on src/transcription/transcription.py:
```diff
@@ -0,0 +1,140 @@
+import os
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from pathlib import Path
+from typing import Any, Dict, List
```
Any, Dict не используется

KORALLLL commented on src/transcription/transcription.py:
```diff
@@ -0,0 +1,140 @@
+from typing import Any, Dict, List
+
+import gigaam 
+import pandas as pd
```
не используется

KORALLLL commented on src/transcription/transcription.py:
```diff
@@ -0,0 +1,140 @@
+import pandas as pd
+import torch
+import torchaudio
+import yaml
```
не используется

KORALLLL commented on src/yofication/yofication.py:
```diff
@@ -0,0 +1,88 @@
+import argparse
+import re
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from typing import List
```
не используется

KORALLLL replied:
воу, а почему всё йофикацию убрал?

KORALLLL commented on src/collate.py:
```diff
@@ -0,0 +1,69 @@
+        "-p",
+        type=str,
+        help="Path to dataset directory"
+    )
```
в каких случаях используется

NikiPshg replied:
если файл запускается не через конфиг а через командную строку все sh файлы придерживались тактой структуры

KORALLLL commented on src/utils.py:
```diff
@@ -0,0 +1,108 @@
+from pathlib import Path
+from typing import List
+import os
```
не используется

KORALLLL replied:
удали os, актуально

KORALLLL commented on src/recovery_from_meta.py:
```diff
@@ -0,0 +1,106 @@
+    parser = argparse.ArgumentParser(description="Create Dataset using meta")
+    parser.add_argument("--parquet_path", "-p", type=str, help="Path to parquet file")
+    parser.add_argument("--config_path", "-c", type=str, help="Path to config file")
+    parser.add_argument("--podcasts_path", "-d", type=str, help="Path to podcasts directory")
```
когда конфиг паз и количество воркеров используются

KORALLLL replied:
понял

KORALLLL commented on src/recovery_from_meta.py:
```diff
@@ -0,0 +1,106 @@
+import torchaudio
+from tqdm import tqdm
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from typing import List, Optional, Tuple
```
все не используется

KORALLLL commented on README.md:
````diff
@@ -0,0 +1,132 @@
+## Prerequisites
+
+```bash
+sudo apt update && sudo apt install ffmpeg
````
Такого ffmpeg достаточно? Если да, то ок

NikiPshg replied:
есть момент с fasterwhisper думаю добавить информацию об этом

KORALLLL replied:
Добавил?

KORALLLL commented on README.md:
````diff
@@ -0,0 +1,132 @@
+## Download Dataset Using Existing Metadata
+
+if you want to use all the metadata that we have, download balalaika parquet and balalaika.pkl and then run
+```bash
+bash use_meta.sh
+```
````
```bash
bash use_meta_{n}h.sh
```

KORALLLL replied:
ляяя, я тупанул и не увидел... Верни прошлую версию этого файла, пжжпжпжп

KORALLLL commented on README.md:
```diff
@@ -0,0 +1,132 @@
+## Running the Pipeline
+
+### Basic Scenario (BASE) 
+if you want to post the data yourself
```
что это значит?

KORALLLL commented on README.md:
Создай issue со следующим содержанием - добавить скрипты для получения расширения имеющегося датасета.

KORALLLL replied:
а ещё внизу добавь заголовок, типа, наша статья. и напиши, что просим вас нас цитировать.

KORALLLL replied:
А ещё нигде не написано о том, что мету можно скачать с хаггинга.

KORALLLL replied:
актуально

KORALLLL commented on use_meta.sh:
```diff
@@ -0,0 +1,11 @@
+# wget ... (download meta .pickle )
+
+PODCASTS_PATH="../Balalaika"
+PICKLE_PATH="Balalaika.pkl"
```
откуда этот файл берётся? и почему он пикл, а не .txt?

KORALLLL replied:
мб добавить возможность, чтобы люди не указывали путь до файл с url, а он автоматически цеплялся с hugging face?

KORALLLL replied:
блин, давай знаешь как сделаем? там же у wget можно указать параметры выходной дирки, вот давай их укажаем? И из этой дирки уже будем открывать аудио и пикл?

KORALLLL reviewed (changes requested):
жду изменения

NikiPshg committed:
docs: update README, clean imports, update .gitignore

NikiPshg committed:
chore: minor updates

NikiPshg committed:
feat: migrate to UV + dual-env system

- Add UV support for faster package management
- Introduce dev/user environment separation
- Update README with setup instructions

NikiPshg committed:
Fix bug: - adding phonemes to the final parquet

NikiPshg committed:
feat: add timestamps and CTC+LM (SOTA) transcription

NikiPshg committed:
fix(transcription): minor bug fixes

NikiPshg committed:
feat(collate): optimize data processing speed
feat(transcription): add CTC+LM (with timestamps) decoding support
feat(device): remove CPU compatibility (GPU-only)
fix(classification): resolve 'empty audio' bug

NikiPshg committed:
Merge pull request #2 from mtuciru/timestamps_lm

Timestamps lm

NikiPshg committed:
update readme

KORALLLL commented on README.md:
```diff
@@ -0,0 +1,197 @@
+- [NISQA](https://github.com/gabrielmittag/NISQA)  – Audio quality assessment.
+- [GigaAM](https://github.com/salute-developers/GigaAM)  – ASR.
+- [ruAccent](https://github.com/Den4ikAI/ruaccent)  – Accent restoration.
+- [RUPynct](https://huggingface.co/RUPunct/RUPunct_big)  – Punctuation restoration.
```
RUPunct

NikiPshg committed:
Merge pull request #3 from mtuciru/timestamps_lm

update readme

KORALLLL reviewed (changes requested):
минорные изменения жду

NikiPshg committed:
update readme

NikiPshg committed:
update readme

NikiPshg committed:
feat(use_meta): save datasets in repo directory
docs: update README
chore: update requirements_user.txt

NikiPshg committed:
feat(use_meta): save datasets in repo directory

NikiPshg committed:
Update README.md

KORALLLL reviewed (approved):
ok
